### PR TITLE
Use mode private, fixes an issue where malformed json was coming back from the ADAL cache

### PR DESCRIPTION
**Pull request by iambmelt**
This PR corrects a bug discovered in https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1616 that when different `MODES` where passed to `SharedPreferencesFileManager` that multiple "singletons" would exist. The resulting bug was inconsistency in the different references of the cache as to what the underlying file on disk contained.

This PR addresses this by doing the following:
- Where `-1` is passed to the cache, treat this as `MODE_PRIVATE` and give it a cache key name that matches this behavior
- Where `MODE_PRIVATE` is passed to the cache, do the same as above
- Expose a new static constructor that defaults to `MODE_PRIVATE`

### CODE REVIEW

**Review by AdamBJohnsonx — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ protected void initializeSharedPreferencesFileManager(final String fileName) {
                 SharedPreferencesFileManager.getSharedPreferences(
                         getContext(),
                         fileName,
-                        -1,
+                        Context.MODE_PRIVATE,
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: hmm...isn't it same as per this code here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/10ccbf9fa3236a980500da68c10063153dd68ace/common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java#L163

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: I believe that's what's being called internally here

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: It should be.  There's no reason not to be explicit, though.  I was worried that I'd done something really strange that was working and shouldn't have been.

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: Given that [`testUpdateTokenCacheUsesResultAuthority` fails in this PR](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1616) until I remove this line, it is seemingly not the same -- not very obvious what's going on here...

@AdamBJohnsonx as FYI....

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: Ahhh I figured this out... I think what's happening here is that the line above[ where the `key` gets created doesn't account for this behavior](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/10ccbf9fa3236a980500da68c10063153dd68ace/common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java#L82)... so unless the cache is initialized with the same args everywhere you end up with duplicates "singletons"

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: Realistically, do we just want to remove the option for different modes?

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:122]: @AdamBJohnsonx Probably, but would be take a breaking change to do it?

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java]: Made this change in an additive way...

**Review by AdamBJohnsonx — APPROVED** [review]

File: common/src/androidTest/java/com/microsoft/identity/common/SharedPreferencesFileManagerTests.java [hunk 2]
```diff
@@ -45,6 +45,8 @@
 import static junit.framework.Assert.assertFalse;
 import static junit.framework.Assert.assertTrue;
 
+import android.content.Context;
```
Comment (shahzaibj) [inline common/src/androidTest/java/com/microsoft/identity/common/SharedPreferencesFileManagerTests.java:48]: Is this import needed?

File: common/src/main/java/com/microsoft/identity/common/internal/cache/HelloCache.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public static void setIsEnabled(final boolean value) {
     public HelloCache(final @NonNull Context context,
                       final @NonNull String protocolName,
                       final @NonNull String targetAppPackageName) {
-        mFileManager = SharedPreferencesFileManager.getSharedPreferences(context, SHARED_PREFERENCE_NAME, -1, null);
+        mFileManager = SharedPreferencesFileManager.getSharedPreferences(context, SHARED_PREFERENCE_NAME,null);
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/HelloCache.java:80]:
```suggestion
        mFileManager = SharedPreferencesFileManager.getSharedPreferences(context, SHARED_PREFERENCE_NAME, null);
```

**Review by shahzaibj — APPROVED** [review]

File: common/src/androidTest/java/com/microsoft/identity/common/SharedPreferencesFileManagerTests.java (hunk 2 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/SharedPreferencesFileManagerTests.java]: nope

File: common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java [hunk 4]
```diff
@@ -90,6 +90,20 @@ public static SharedPreferencesFileManager getSharedPreferences(final Context co
...
 
+    /**
+     * Constructs an instance of SharedPreferencesFileManager. Operating mode is always MODE_PRIVATE.
+     *
+     * @param context Interface to global information about an application environment.
+     * @param name The desired {@link android.content.SharedPreferences} file. It will be created if it does not exist.
+     * @param storageHelper The {@link IStorageHelper} to handle encryption/decryption of values.
+     * @return The SharedPreferencesFileManager instance.
+     */
+    public static SharedPreferencesFileManager getSharedPreferences(final Context context,
```
Comment (iamgusain) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java:101]: nit: If we are only going to use this to getSharedPreferences.. we can mark the other one (line78) as deprecated

**Review by iamgusain — APPROVED** [review]
Approved with suggestion

File: common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java (hunk 4 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java:102]: 15b27a5ca2fa926df66223b7f9380c30b810b20c

**Pull request merged**